### PR TITLE
Delete `Praos.Translate` module

### PR DESCRIPTION
**Pull request by jasagredo**

### CODE REVIEW

**Review by jorisdral — APPROVED** [review]

File: ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs [hunk 1]
```diff
@@ -1,22 +1,95 @@
...
+import qualified Ouroboros.Consensus.Protocol.Abstract as Abstract
+import           Ouroboros.Consensus.Protocol.Praos (Praos, PraosState (..))
+import           Ouroboros.Consensus.Protocol.Praos.Views
+                     (LedgerView (lvMaxBodySize, lvMaxHeaderSize, lvProtocolVersion))
+import qualified Ouroboros.Consensus.Protocol.Praos.Views as Views
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos,
+                     TPraosState (tpraosStateChainDepState, tpraosStateLastSlot))
 
 -- | Translate across protocols
 class TranslateProto protoFrom protoTo
```
Comment (amesgen) [inline ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs:31]: Maybe we could even move this class to `oc-cardano`?

**Review by amesgen — APPROVED** [review]

**Comment by jasagredo** [comment]
The PR moved some other things around. Please re-review (no rush) @amesgen @jorisdral

File: ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs [hunk 2]
```diff
@@ -1,22 +1,95 @@
...
+-- - They share the same ADDRHASH algorithm
+-- - They share the same DSIGN verification keys
+-- - They share the same VRF verification keys
+instance
+  ( HASH c1 ~ HASH c2,
+    ADDRHASH c1 ~ ADDRHASH c2,
+    VerKeyDSIGN c1 ~ VerKeyDSIGN c2,
+    VerKeyVRF c1 ~ VerKeyVRF c2
+  ) =>
+  TranslateProto (TPraos c1) (Praos c2)
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs]: Done!

File: ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs (hunk 1 above)
Comment (jasagredo) [reply ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Translate.hs]: I moved it to `Ouroboros.Consensus.Protocol.Abstract` :thinking:

**Review by amesgen — APPROVED** [review]

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Protocol/Abstract.hs [hunk 3]
```diff
@@ -1,7 +1,10 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Protocol/Abstract.hs]: I added a Proxy, just a couple of changes and it was done 👌

**Pull request merged**